pysph/base/opencl.py

```diff
@@ -35,6 +35,47 @@ def set_queue(q):
     _queue = q
 
 
+class DeviceArray(cl.array.Array):
+    def __init__(self, queue, dtype, n=0):
+        self.queue = queue
+        self.length = n
+        if n == 0:
+            n = 16
+        self.alloc = n
+        self.dtype = dtype
+        self._array = cl.array.empty(queue, n, dtype)
```

Comment:
Why is this alone with an `_array`?

Reply:
Instead maybe this should be `_data` and the property should really be `array` because that is what it is, right?  This `_array` is just the block of data allocated but the array is a suitable view.